### ISSUE TITLE
Constant autoloading deprecation warning with Sidekiq Ent periodic jobs

### ISSUE THREAD

**Issue by christos** [mperham/sidekiq#4271]
Ruby version: 2.5.5
Sidekiq / Pro / Enterprise version(s): Sidekiq Pro 5.0.0 / Sidekiq Enterprise 2.0.0

On Rails 6.0 final, I am seeing the same issue as mentioned in https://github.com/mperham/sidekiq/issues/4192

```
 DEPRECATION WARNING: Initialization autoloaded the constants Billing and Billing::GoogleInvoicesImportWorker.

 Being able to do this is deprecated. Autoloading during initialization is going
 to be an error condition in future versions of Rails.

 Reloading does not reboot the application, and therefore code executed during
 initialization does not run again. So, if you reload Billing, for example,
 the expected changes won't be reflected in that stale Module object.

 `config.autoloader` is set to `classic`. These autoloaded constants would have been unloaded if `config.autoloader` had been set to `:zeitwerk`.

 Please, check the "Autoloading and Reloading Constants" guide for solutions.
  (called from <top (required)> at /Users/christos/Github/lingokids/api/config/environment.rb:7)
```

My Sidekiq initialiser is 

```
require "sidekiq/pro/web"
require "sidekiq_instrumentation_middleware"

redis_config = {
  url: ENV.fetch("API__REDIS_SIDEKIQ_URL"),
  driver: "hiredis",
  network_timeout: 5
}

Sidekiq::Web.set :session_secret, Rails.application.credentials[:secret_key_base]
Sidekiq::Web.set :sessions,       Rails.configuration.session_options

# https://github.com/mperham/sidekiq/wiki/Pro-Reliability-Client
Sidekiq::Client.reliable_push! unless Rails.env.test?

Sidekiq.configure_server do |config|
  config.redis = redis_config
  config.super_fetch!
  config.server_middleware do |chain|
    chain.add SidekiqInstrumentationMiddleware
  end
  config.periodic do |mgr|
    # see any crontab reference for the first argument
    # e.g. http://www.adminschoice.com/crontab-quick-reference
    # or   https://crontab.guru/
    # mgr.register('0 * * * *', SomeHourlyWorkerClass)
    # mgr.register('* * * * *', SomeWorkerClass, retry: 2, queue: 'foo')
    # mgr.register(cron_expression, worker_class, job_options={})

    mgr.register("30 7 * * *", Billing::GoogleInvoicesImportWorker)
  end
```

My temp fix is to use Strings for the worker names, as suggested in the quoted issue.

**Comment by christos** [comment]
I've also tried enabling `zeitwerk` as the autoloader. Same warning minus the mention of the `classic` autoloader being enabled.

**Comment by mperham** [comment]
Can you extract all this into a minimal app which reproduces the warning?

**Comment by christos** [comment]
I don't seem to be able to reproduce it in a new Rails 6 app.

I've reduced the Sidekiq initialiser to the following with an empty `BasicWorker` but still get the same warning.

```ruby
Sidekiq.configure_server do |config|
  config.periodic do |mgr|
    mgr.register("30 7 * * *", BasicWorker)
  end
end
```

They only extra debug info I've gotten so far is to enable `ActiveSupport::Deprecation` debugging and add some tracing `puts` to see which initialiser is loaded before the deprecation warning is fired:

<details>
<summary>Trace of initialisers loading</summary>
<pre>
$ sidekiq -C config/sidekiq.yml
Initializer: set_load_path 		 Context Class: Sidekiq::Rails
Initializer: set_load_path 		 Context Class: ActionView::Railtie
Initializer: set_load_path 		 Context Class: Jquery::Rails::Engine
Initializer: set_load_path 		 Context Class: PaperTrail::Rails::Engine
Initializer: set_load_path 		 Context Class: PopperJs::Engine
Initializer: set_load_path 		 Context Class: Bootstrap::Rails::Engine
Initializer: set_load_path 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: set_load_path 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: set_load_path 		 Context Class: Kaminari::Engine
Initializer: set_load_path 		 Context Class: MyApp::Application
Initializer: set_autoload_paths 		 Context Class: Sidekiq::Rails
Initializer: set_autoload_paths 		 Context Class: ActionView::Railtie
Initializer: set_autoload_paths 		 Context Class: Jquery::Rails::Engine
Initializer: set_autoload_paths 		 Context Class: PaperTrail::Rails::Engine
Initializer: set_autoload_paths 		 Context Class: PopperJs::Engine
Initializer: set_autoload_paths 		 Context Class: Bootstrap::Rails::Engine
Initializer: set_autoload_paths 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: set_autoload_paths 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: set_autoload_paths 		 Context Class: Kaminari::Engine
Initializer: set_autoload_paths 		 Context Class: MyApp::Application
Initializer: set_eager_load_paths 		 Context Class: Sidekiq::Rails
Initializer: set_eager_load_paths 		 Context Class: ActionView::Railtie
Initializer: set_eager_load_paths 		 Context Class: Jquery::Rails::Engine
Initializer: set_eager_load_paths 		 Context Class: PaperTrail::Rails::Engine
Initializer: set_eager_load_paths 		 Context Class: PopperJs::Engine
Initializer: set_eager_load_paths 		 Context Class: Bootstrap::Rails::Engine
Initializer: set_eager_load_paths 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: set_eager_load_paths 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: set_eager_load_paths 		 Context Class: Kaminari::Engine
Initializer: set_eager_load_paths 		 Context Class: MyApp::Application
Initializer: add_routing_paths 		 Context Class: Sidekiq::Rails
Initializer: add_routing_paths 		 Context Class: ActionView::Railtie
Initializer: add_routing_paths 		 Context Class: Jquery::Rails::Engine
Initializer: add_routing_paths 		 Context Class: PaperTrail::Rails::Engine
Initializer: add_routing_paths 		 Context Class: PopperJs::Engine
Initializer: add_routing_paths 		 Context Class: Bootstrap::Rails::Engine
Initializer: add_routing_paths 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: add_routing_paths 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: add_routing_paths 		 Context Class: Kaminari::Engine
Initializer: add_routing_paths 		 Context Class: MyApp::Application
Initializer: add_locales 		 Context Class: Sidekiq::Rails
Initializer: add_locales 		 Context Class: ActionView::Railtie
Initializer: add_locales 		 Context Class: Jquery::Rails::Engine
Initializer: add_locales 		 Context Class: PaperTrail::Rails::Engine
Initializer: add_locales 		 Context Class: PopperJs::Engine
Initializer: add_locales 		 Context Class: Bootstrap::Rails::Engine
Initializer: add_locales 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: add_locales 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: add_locales 		 Context Class: Kaminari::Engine
Initializer: add_locales 		 Context Class: MyApp::Application
Initializer: add_view_paths 		 Context Class: Sidekiq::Rails
Initializer: add_view_paths 		 Context Class: ActionView::Railtie
Initializer: add_view_paths 		 Context Class: Jquery::Rails::Engine
Initializer: add_view_paths 		 Context Class: PaperTrail::Rails::Engine
Initializer: add_view_paths 		 Context Class: PopperJs::Engine
Initializer: add_view_paths 		 Context Class: Bootstrap::Rails::Engine
Initializer: add_view_paths 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: add_view_paths 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: add_view_paths 		 Context Class: Kaminari::Engine
Initializer: add_view_paths 		 Context Class: MyApp::Application
Initializer: load_environment_config 		 Context Class: Sidekiq::Rails
Initializer: load_environment_config 		 Context Class: ActionView::Railtie
Initializer: load_environment_config 		 Context Class: Jquery::Rails::Engine
Initializer: load_environment_config 		 Context Class: PaperTrail::Rails::Engine
Initializer: load_environment_config 		 Context Class: PopperJs::Engine
Initializer: load_environment_config 		 Context Class: Bootstrap::Rails::Engine
Initializer: load_environment_config 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: load_environment_config 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: load_environment_config 		 Context Class: Kaminari::Engine
Initializer: load_environment_config 		 Context Class: MyApp::Application
Initializer: load_environment_hook 		 Context Class: MyApp::Application
Initializer: load_active_support 		 Context Class: MyApp::Application
Initializer: set_eager_load 		 Context Class: MyApp::Application
Initializer: initialize_logger 		 Context Class: MyApp::Application
Initializer: initialize_cache 		 Context Class: MyApp::Application
Initializer: initialize_dependency_mechanism 		 Context Class: MyApp::Application
Initializer: bootstrap_hook 		 Context Class: MyApp::Application
Initializer: set_secrets_root 		 Context Class: MyApp::Application
Initializer: active_support.set_authenticated_message_encryption 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.reset_all_current_attributes_instances 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.deprecation_behavior 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.initialize_time_zone 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.initialize_beginning_of_week 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.require_master_key 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.set_configs 		 Context Class: ActiveSupport::Railtie
Initializer: active_support.set_hash_digest_class 		 Context Class: ActiveSupport::Railtie
Initializer: action_dispatch.configure 		 Context Class: ActionDispatch::Railtie
Initializer: active_model.secure_password 		 Context Class: ActiveModel::Railtie
Initializer: active_model.i18n_customize_full_message 		 Context Class: ActiveModel::Railtie
Initializer: global_id 		 Context Class: GlobalID::Railtie
Initializer: active_job.logger 		 Context Class: ActiveJob::Railtie
Initializer: active_job.custom_serializers 		 Context Class: ActiveJob::Railtie
Initializer: active_job.set_configs 		 Context Class: ActiveJob::Railtie
Initializer: active_job.set_reloader_hook 		 Context Class: ActiveJob::Railtie
Initializer: action_controller.assets_config 		 Context Class: ActionController::Railtie
Initializer: action_controller.set_helpers_path 		 Context Class: ActionController::Railtie
Initializer: action_controller.parameters_config 		 Context Class: ActionController::Railtie
Initializer: action_controller.set_configs 		 Context Class: ActionController::Railtie
Initializer: action_controller.compile_config_methods 		 Context Class: ActionController::Railtie
Initializer: action_controller.request_forgery_protection 		 Context Class: ActionController::Railtie
Initializer: action_controller.eager_load_actions 		 Context Class: ActionController::Railtie
Initializer: active_record.initialize_timezone 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.logger 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.backtrace_cleaner 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.migration_error 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.database_selector 		 Context Class: ActiveRecord::Railtie
Initializer: Check for cache versioning support 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.check_schema_cache_dump 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.define_attribute_methods 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.warn_on_records_fetched_greater_than 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.set_configs 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.initialize_database 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.log_runtime 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.collection_cache_association_loading 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.set_reloader_hooks 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.set_executor_hooks 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.add_watchable_files 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.clear_active_connections 		 Context Class: ActiveRecord::Railtie
Initializer: active_record.set_filter_attributes 		 Context Class: ActiveRecord::Railtie
Initializer: action_mailer.logger 		 Context Class: ActionMailer::Railtie
Initializer: action_mailer.set_configs 		 Context Class: ActionMailer::Railtie
Initializer: action_mailer.set_autoload_paths 		 Context Class: ActionMailer::Railtie
Initializer: action_mailer.compile_config_methods 		 Context Class: ActionMailer::Railtie
Initializer: action_mailer.eager_load_actions 		 Context Class: ActionMailer::Railtie
Initializer: set_default_precompile 		 Context Class: Sprockets::Railtie
Initializer: quiet_assets 		 Context Class: Sprockets::Railtie
Initializer: rails-i18n 		 Context Class: RailsI18n::Railtie
Initializer: request_store.insert_middleware 		 Context Class: RequestStore::Railtie
Initializer: setup_sass 		 Context Class: SassC::Rails::Railtie
Initializer: setup_compression 		 Context Class: SassC::Rails::Railtie
Initializer: kaminari 		 Context Class: Kaminari::Railtie
Initializer: roadie-rails.setup 		 Context Class: Roadie::Rails::Railtie
Initializer: hashie.configure_logger 		 Context Class: Hashie::Railtie
Initializer: raven.use_rack_middleware 		 Context Class: Raven::Rails
Initializer: raven.action_controller 		 Context Class: Raven::Rails
Initializer: raven.action_view 		 Context Class: Raven::Rails
Initializer: raven.active_job 		 Context Class: Raven::Rails
Initializer: scout_apm.start 		 Context Class: ScoutApm::Railtie
Initializer: better_errors.configure_rails_initialization 		 Context Class: BetterErrors::Railtie
Initializer: prepend_helpers_path 		 Context Class: Sidekiq::Rails
Initializer: prepend_helpers_path 		 Context Class: ActionView::Railtie
Initializer: prepend_helpers_path 		 Context Class: Jquery::Rails::Engine
Initializer: prepend_helpers_path 		 Context Class: PaperTrail::Rails::Engine
Initializer: prepend_helpers_path 		 Context Class: PopperJs::Engine
Initializer: prepend_helpers_path 		 Context Class: Bootstrap::Rails::Engine
Initializer: prepend_helpers_path 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: prepend_helpers_path 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: prepend_helpers_path 		 Context Class: Kaminari::Engine
Initializer: prepend_helpers_path 		 Context Class: MyApp::Application
Initializer: load_config_initializers 		 Context Class: Sidekiq::Rails
Initializer: load_config_initializers 		 Context Class: ActionView::Railtie
Initializer: load_config_initializers 		 Context Class: Jquery::Rails::Engine
Initializer: load_config_initializers 		 Context Class: PaperTrail::Rails::Engine
Initializer: load_config_initializers 		 Context Class: PopperJs::Engine
Initializer: load_config_initializers 		 Context Class: Bootstrap::Rails::Engine
Initializer: load_config_initializers 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: load_config_initializers 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: load_config_initializers 		 Context Class: Kaminari::Engine
Initializer: load_config_initializers 		 Context Class: MyApp::Application
Sentry configuration not set
2019-09-16T10:52:52.256Z pid=73349 tid=ovi1jmftp INFO: Booting Sidekiq 6.0.0 with redis options {:id=>"Sidekiq-server-PID-73349", :url=>nil}
2019-09-16T10:52:52.256Z pid=73349 tid=ovi1jmftp INFO: Booting Sidekiq 6.0.0 with redis options {:url=>"redis://127.0.0.1:6379/0", :driver=>"hiredis", :network_timeout=>5, :id=>"Sidekiq-server-PID-73349"}
Initializer: engines_blank_point 		 Context Class: Sidekiq::Rails
Initializer: engines_blank_point 		 Context Class: ActionView::Railtie
Initializer: engines_blank_point 		 Context Class: Jquery::Rails::Engine
Initializer: engines_blank_point 		 Context Class: PaperTrail::Rails::Engine
Initializer: engines_blank_point 		 Context Class: PopperJs::Engine
Initializer: engines_blank_point 		 Context Class: Bootstrap::Rails::Engine
Initializer: engines_blank_point 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: engines_blank_point 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: engines_blank_point 		 Context Class: Kaminari::Engine
Initializer: engines_blank_point 		 Context Class: MyApp::Application
Initializer: append_assets_path 		 Context Class: Sidekiq::Rails
Initializer: sidekiq.active_job_integration 		 Context Class: Sidekiq::Rails
Initializer: append_assets_path 		 Context Class: ActionView::Railtie
Initializer: action_view.embed_authenticity_token_in_remote_forms 		 Context Class: ActionView::Railtie
Initializer: action_view.form_with_generates_remote_forms 		 Context Class: ActionView::Railtie
Initializer: action_view.form_with_generates_ids 		 Context Class: ActionView::Railtie
Initializer: action_view.default_enforce_utf8 		 Context Class: ActionView::Railtie
Initializer: action_view.finalize_compiled_template_methods 		 Context Class: ActionView::Railtie
Initializer: action_view.logger 		 Context Class: ActionView::Railtie
Initializer: action_view.set_configs 		 Context Class: ActionView::Railtie
Initializer: action_view.caching 		 Context Class: ActionView::Railtie
Initializer: action_view.per_request_digest_cache 		 Context Class: ActionView::Railtie
Initializer: action_view.setup_action_pack 		 Context Class: ActionView::Railtie
Initializer: action_view.collection_caching 		 Context Class: ActionView::Railtie
Initializer: append_assets_path 		 Context Class: Jquery::Rails::Engine
Initializer: append_assets_path 		 Context Class: PaperTrail::Rails::Engine
Initializer: paper_trail.initialisation 		 Context Class: PaperTrail::Rails::Engine
Initializer: append_assets_path 		 Context Class: PopperJs::Engine
Initializer: popper_js.assets 		 Context Class: PopperJs::Engine
Initializer: append_assets_path 		 Context Class: Bootstrap::Rails::Engine
Initializer: bootstrap.assets 		 Context Class: Bootstrap::Rails::Engine
Initializer: append_assets_path 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: tabler.assets 		 Context Class: Tabler::Rubygem::Rails::Engine
Initializer: append_assets_path 		 Context Class: Jquery::Ui::Rails::Engine
Initializer: append_assets_path 		 Context Class: Kaminari::Engine
Initializer: append_assets_path 		 Context Class: MyApp::Application
Initializer: add_generator_templates 		 Context Class: MyApp::Application
Initializer: ensure_autoload_once_paths_as_subset 		 Context Class: MyApp::Application
Initializer: warn_if_autoloaded 		 Context Class: MyApp::Application
DEPRECATION WARNING: Initialization autoloaded the constants Billing and Billing::GoogleInvoicesImportWorker.

Being able to do this is deprecated. Autoloading during initialization is going
to be an error condition in future versions of Rails.

Reloading does not reboot the application, and therefore code executed during
initialization does not run again. So, if you reload Billing, for example,
the expected changes won't be reflected in that stale Module object.

These autoloaded constants have been unloaded.

Please, check the "Autoloading and Reloading Constants" guide for solutions.
 (called from <top (required)> at /Users/christos/Github/lingokids/api/config/environment.rb:7)
/Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/railties-6.0.0/lib/rails/initializable.rb:32:in `instance_exec'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/railties-6.0.0/lib/rails/initializable.rb:32:in `run'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/railties-6.0.0/lib/rails/initializable.rb:62:in `block in run_initializers'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:228:in `block in tsort_each'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:431:in `each_strongly_connected_component_from'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:349:in `block in each_strongly_connected_component'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:347:in `each'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:347:in `call'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:347:in `each_strongly_connected_component'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:226:in `tsort_each'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/2.5.0/tsort.rb:205:in `tsort_each'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/railties-6.0.0/lib/rails/initializable.rb:60:in `run_initializers'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/railties-6.0.0/lib/rails/application.rb:363:in `initialize!'
  /Users/christos/Github/lingokids/api/config/environment.rb:7:in `<top (required)>'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/activesupport-6.0.0/lib/active_support/dependencies.rb:325:in `require'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/activesupport-6.0.0/lib/active_support/dependencies.rb:325:in `block in require'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/activesupport-6.0.0/lib/active_support/dependencies.rb:291:in `load_dependency'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/activesupport-6.0.0/lib/active_support/dependencies.rb:325:in `require'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/sidekiq-6.0.0/lib/sidekiq/cli.rb:248:in `boot_system'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/sidekiq-6.0.0/lib/sidekiq/cli.rb:37:in `run'
  /Users/christos/.rbenv/versions/2.5.5/lib/ruby/gems/2.5.0/gems/sidekiq-6.0.0/bin/sidekiq:12:in `<top (required)>'
  ./.git/safe/../../bin/sidekiq:29:in `load'
  ./.git/safe/../../bin/sidekiq:29:in `<main>'
Initializer: let_zeitwerk_take_over 		 Context Class: MyApp::Application
Initializer: add_builtin_route 		 Context Class: MyApp::Application
Initializer: setup_default_session_store 		 Context Class: MyApp::Application
Initializer: build_middleware_stack 		 Context Class: MyApp::Application
Initializer: define_main_app_helper 		 Context Class: MyApp::Application
Initializer: add_to_prepare_blocks 		 Context Class: MyApp::Application
Initializer: run_prepare_callbacks 		 Context Class: MyApp::Application
Initializer: eager_load! 		 Context Class: MyApp::Application
Initializer: finisher_hook 		 Context Class: MyApp::Application
Initializer: configure_executor_for_concurrency 		 Context Class: MyApp::Application
Initializer: set_routes_reloader_hook 		 Context Class: MyApp::Application
Initializer: set_clear_dependencies_hook 		 Context Class: MyApp::Application
Initializer: disable_dependency_loading 		 Context Class: MyApp::Application

                                   /\.                            /\
        L -      --- -- ---,,-.  ./.: 8.    .-----8p--.  .---.s,-'.:`--+
 ,8b  8i     ,F.oSSS"Sbo.o.s8 .\/.:'.oD \  /.:::.dS8b .\/ .: o8o ::::::|
 `4P,oP   ,o8;(88(`.:.4Y8 `"'::'.:."8DD:`\/ `:::.`888 :.db.::`P' `:'  `:
   o8P, .o8P   48bo.`:."`.dS`:'.oSSbDD8.`.oSSbo.`.888::oOOo oooo `.oSSbo o/
  d888Ld8P'  `:."88888o.``888 d88',`488 d8('.`88b:88( .8P'.:`888 d88P `88P
  88 8P8P     .`.:.`"Y88b 888 888 O 84P 888bod8P'.888o88 .:::888 88( O 888
 dP'`Y8P      db .:::.)8P 888 888 " 888b888'....: 888`.88b.::888 88b " 88(
T   ;88i      4P"888888P'o888o.Y8bod88P"`Y8bod8P d888b..488od888b V8bod888o
     88'                  48P::`odb.dSo,, ."P'.roy<sac>` ""'.`.o.`.."S"888.
     `8                    ".:'d888P'."S8b.`.::... ...' `.:::.d8b ::'  dP'
     .8                    `  '"888 ::.)88\odbo d8b`.odSso.::.888.::."488"
     .8                       `:888bod88P'.`888P"8Pd88'`488b::888::::.`8\
     P                        .:888 "P'..:::88(.:."88( O )88::48P::'/'/\ \
                             /\ 888`::./\`::888:::,888.".888::.o./\<:<  > >
 -   -  -- --- ------.  .---'  d888bo./  `-d888b.`:`Y8SbS8P'--48P'  \ \/ /
                      \/       `8P':/        "   \/   """      "     \/\/
                                P'\/                                    /<>\

2019-09-16T10:52:52.667Z pid=73349 tid=ovi1jmftp INFO: Running in ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-darwin18]
2019-09-16T10:52:52.667Z pid=73349 tid=ovi1jmftp INFO: Sidekiq Pro 5.0.0 / Sidekiq Enterprise 2.0.0, commercially licensed.
2019-09-16T10:52:52.680Z pid=73349 tid=ovi1jmftp INFO: Starting processing, hit Ctrl-C to stop
2019-09-16T10:52:52.681Z pid=73349 tid=ovi297ct9 INFO: SuperFetch activated
2019-09-16T10:52:52.692Z pid=73349 tid=ovi2989r5 INFO: Managing 3 periodic jobs
2019-09-16T10:52:52.695Z pid=73349 tid=ovi2989r5 INFO: Gained leadership of the cluster
</pre>
</details>

**Comment by christos** [comment]
I think I've figured it out. This is not specific to Sidekiq.

**Any initialiser** that references a constant in the autoload path produces the same warning.

What is happening above is that my workers live under `app/workers` which is by default an auto-loaded path, and hence the warning.

The following initialiser also causes the warning:

```ruby
# config/initializers/dummy.rb

BasicWorker # Defined in app/workers/basic_worker.rb
```

I guess it is expected behaviour? 

Possible workaround?
 
* Use a String for the class name and mention it in Sidekiq's docs?
* Remove `app/workers` from the autoload path or tell users to use `lib/periodic_workers` or similar place for periodic workers?
* Delay Sidekiq initialisation by wrapping it `ActiveSupport.on_load` (not sure this would work)

**Comment by mperham** [comment]
Take a look at the diff of config/application.rb between your app and the minimal app. I need a reproduction app so I can redesign the periodic API and verify it works.

**Comment by christos** [comment]
Scratch that. It is something with my app that causes this. I can't reproduce my assumption above with a simple Rails app. 

I'll keep digging. Feel free to close and I can re-open if I find the culprit.

**Comment by christos** [comment]
OK. Managed to reproduce the bug in a vanilla Rails app. Nothing to do with Sidekiq; it was just triggered from having the Sidekiq initialiser reference the worker class.

Here is the reproduction in case you are interested https://github.com/christos/rails-6-sandbox (Not even sure who to raise this issue with, probably Rails)

Thanks for the quick response and the suggestion to dissect `application.rb`. Closing this ticket.

**Comment by mperham** [comment]
It's not a Rails issue; I need to redesign Enterprise's periodic configuration so it doesn't require class loading in the initializer.  You'll just do this instead:

```ruby
mgr.register("30 7 * * *", "Billing::GoogleInvoicesImportWorker")
```

**Comment by mperham** [comment]
Thank you for the app, this is fixed and will be available in Enterprise 2.0.1.

**Comment by Anees-devsinc** [comment]
any workaround, this is same for non-enterprise sidekiqs
